client/apiclient.go

```diff
@@ -47,13 +50,82 @@ type TLSConfig struct {
 }
 
 // NewAPIClient gets an API client for a product.
-func NewAPIClient(product, baseURL string, headers map[string]string) *APIClient {
+func NewAPIClient(product, baseURL string, headers map[string]string, t TLSConfig) (*APIClient, error) {
+	tlsClientConfig, err := createTLSClientConfig(t)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsClientConfig,
+		},
+	}
+
 	return &APIClient{
 		Product: product,
 		BaseURL: baseURL,
 		headers: headers,
-		http:    http.DefaultClient,
+		http:    client,
+	}, nil
+}
+
+func createTLSClientConfig(t TLSConfig) (*tls.Config, error) {
+	tlsClientConfig := &tls.Config{}
+
+	// If the input TLSConfig object is default, we do not need to update tlsClientConfig.
+	if reflect.DeepEqual(t, TLSConfig{}) {
+		return tlsClientConfig, nil
 	}
+
+	if t.CACert != "" || len(t.CACertBytes) != 0 || t.CAPath != "" {
+		rootConfig := &rootcerts.Config{
+			CAFile:        t.CACert,
+			CACertificate: t.CACertBytes,
+			CAPath:        t.CAPath,
+		}
+		if err := rootcerts.ConfigureTLS(tlsClientConfig, rootConfig); err != nil {
+			return nil, err
```

Comment:
personal preference: returning a `nil` pointer, even alongside an error, always makes me `panic` a little anticipatorily, so I usually prefer to return an empty thing 

```suggestion
			return &tls.Config{}, err
```

or however far along it has been built so far

```suggestion
			return tlsClientConfig, err
```


on the flip side, a full-on panic sure would let us know if we're not handling that error somewhere 😜

Reply:
From a quick scan through the code base, it looks like it is more common (today) to `return nil, err`. I wonder if this is something that we should chat about as a group and come up with a preferred style on? I can see the reasoning for the concern (the "panic" as it were! 😆 ), but it doesn't look like it's common in hcdiag today. (/cc @mkcp for thoughts too!)